spec/controllers/idv/personal_key_controller_spec.rb

```diff
@@ -52,81 +54,175 @@ def assert_personal_key_generated_for_profiles(*profile_pii_pairs)
       idv_session.address_verification_mechanism = 'phone'
       idv_session.user_phone_confirmation = true
       idv_session.vendor_phone_confirmation = true
+      idv_session.gpo_code_verified = false
     when 'gpo'
       idv_session.address_verification_mechanism = 'gpo'
       idv_session.user_phone_confirmation = false
       idv_session.vendor_phone_confirmation = false
+      idv_session.gpo_code_verified = true
+    when nil
+      idv_session.address_verification_mechanism = nil
+      idv_session.user_phone_confirmation = nil
+      idv_session.vendor_phone_confirmation = nil
     else
       raise 'invalid address_verification_mechanism'
     end
 
+    idv_session.threatmetrix_review_status = threatmetrix_review_status
+
     if mint_profile_from_idv_session
       idv_session.create_profile_from_applicant_with_password(password)
     end
   end
 
+  describe '#step_info' do
+    let(:step_info) do
+      controller.class.step_info
+    end
+
+    describe '#undo' do
```

Comment:
Good idea to test #undo_step. We should probably add that to all the controllers...
```suggestion
    describe '#undo_step' do
```